src/components/DomainSearchResultsDisplay.tsx

```diff
@@ -46,3 +47,27 @@ export const DomainSearchResultsDisplay: React.FC<DomainSearchResultProps> = ({
     </div>
   );
 };
+
+function mergeRecordsByValue(records: RecordWithSelector[]): RecordWithSelector[] {
+  const valueMap = new Map<string, RecordWithSelector>();
+
+  records.forEach((record) => {
+    if (valueMap.has(record.value)) {
+      const existing = valueMap.get(record.value)!;
```

Comment:
This will merge all records that have the same `value`, even if the other fields like domain and selector are different. What if there are two different domain-selector-pairs with the same key?
I mean, this _should_ not be the case, but maybe someone has misconfigured their domain and uses someone else's key? Don't we want to reflect that in the UI?
So my idea is that we should only merge if also domain and selector are the same. Not sure about sourceIdentifier (called "Selector origin" and "Key origin" in the UI)

Reply:
Update: It seems that reused keys are very common. We even have a statistics tool for that in `dkim_key_reuse_statistics()` in `src/util/statistics.py` which I had forgot about :)
I have a file with statistics and the most common DKIM key is used on more than 40 000 domains!